tests/python/pants_test/backend/python/tasks2/test_setup_py.py

```diff
@@ -11,33 +11,125 @@
 from textwrap import dedent
 
 from mock import Mock
+from pex.package import Package
 from twitter.common.collections import OrderedSet
 from twitter.common.dirutil.chroot import Chroot
 
+from pants.backend.python.subsystems.python_setup import PythonSetup
 from pants.backend.python.tasks2.setup_py import SetupPy
 from pants.base.exceptions import TaskError
 from pants.build_graph.build_file_aliases import BuildFileAliases
 from pants.build_graph.prep_command import PrepCommand
 from pants.build_graph.resources import Resources
 from pants.build_graph.target import Target
 from pants.fs.archive import TGZ
-from pants.util.contextutil import temporary_dir, temporary_file
+from pants.util.contextutil import environment_as, temporary_dir, temporary_file
 from pants.util.dirutil import safe_mkdir
 from pants_test.backend.python.tasks.python_task_test_base import PythonTaskTestBase
 from pants_test.subsystem.subsystem_util import init_subsystem
 
 
-class TestSetupPy(PythonTaskTestBase):
+class BaeSetupPyTest(PythonTaskTestBase):
```

Comment:
A) Bae? Are we mining Urban Dictionary for names now? ;-)

B) Also, this should probably be SetupPyTestBae, er, Base, for uniformity (we generally assume that anything ending in `Test` tests the thing with the prefix as a name).